packages/amplify_core/lib/src/category/amplify_auth_category.dart

```diff
@@ -95,6 +256,186 @@ class AuthCategory extends AmplifyCategory<AuthPluginInterface> {
+  /// ```dart
+  /// Future<void> _handleSignInResult(SignInResult result) async {
+  ///   switch (result.nextStep.signInStep) {
+  ///     // ···
+  ///     case AuthSignInStep.resetPassword:
+  ///       final resetResult = await Amplify.Auth.resetPassword(
+  ///         username: username,
+  ///       );
+  ///       await _handleResetPasswordResult(resetResult);
+  ///       break;
+  ///     // ···
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="handle-reset-password"?>
+  /// ```dart
+  /// Future<void> _handleResetPasswordResult(ResetPasswordResult result) async {
+  ///   switch (result.nextStep.updateStep) {
+  ///     case AuthResetPasswordStep.confirmResetPasswordWithCode:
+  ///       final codeDeliveryDetails = result.nextStep.codeDeliveryDetails!;
+  ///       _handleCodeDelivery(codeDeliveryDetails);
+  ///       break;
+  ///     case AuthResetPasswordStep.done:
+  ///       safePrint('Successfully reset password');
+  ///       break;
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// If a user tries to sign in before completing their sign up, the [AuthSignInStep.confirmSignUp]
+  /// step will be returned. This is the same step returned from [signUp] and requires a
+  /// call to [confirmSignUp] before re-initiating the sign in flow.
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="handle-confirm-signin-confirm-signup"?>
+  /// ```dart
+  /// Future<void> _handleSignInResult(SignInResult result) async {
+  ///   switch (result.nextStep.signInStep) {
+  ///     // ···
+  ///     case AuthSignInStep.confirmSignUp:
+  ///       // Resend the sign up code to the registered device.
+  ///       final resendResult = await Amplify.Auth.resendSignUpCode(
+  ///         username: username,
+  ///       );
+  ///       _handleCodeDelivery(resendResult.codeDeliveryDetails);
+  ///       break;
+  ///     // ···
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="handle-code"?>
+  /// ```dart
+  /// void _handleCodeDelivery(AuthCodeDeliveryDetails codeDeliveryDetails) {
+  ///   safePrint(
+  ///     'A confirmation code has been sent to ${codeDeliveryDetails.destination}. '
+  ///     'Please check your ${codeDeliveryDetails.deliveryMedium.name} for the code.',
+  ///   );
+  /// }
+  /// ```
+  ///
+  /// When there are no more challenges and the sign in is complete, the next step will
+  /// be [AuthSignInStep.done].
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="handle-confirm-signin-done"?>
+  /// ```dart
+  /// Future<void> _handleSignInResult(SignInResult result) async {
+  ///   switch (result.nextStep.signInStep) {
+  ///     // ···
+  ///     case AuthSignInStep.done:
+  ///       safePrint('Sign in is complete');
+  ///       break;
+  ///     // ···
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// Optionally accepts plugin [options] which allow customizing provider-specific
```

Comment:
The content here is sort of buried by examples. Do you think it is worth putting this above the example?